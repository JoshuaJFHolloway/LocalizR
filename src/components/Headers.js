import React from 'react';
import Header from './Header';
import SubHeader from './SubHeader';
import Font from "./styledComponents/fonts.js";

const Headers = () => {
  return (
    <Font>
      <Header/>
      <img src="https://cdn2.iconfinder.com/data/icons/Flag/134/Spain.png" alt="themeimage" width="150" />
      <SubHeader/>
    </Font>
  );
};

export default Headers;