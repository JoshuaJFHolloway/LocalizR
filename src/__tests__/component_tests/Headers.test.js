import React from 'react';
import { shallow } from 'enzyme';
import Headers from '../../components/Headers';

describe('Headers', () => {
  let headers = shallow(<Headers />);

  describe('Rendering components', () => {

    it('renders correctly', () => {
      expect(headers).toMatchSnapshot();
    });

    it('renders a Header component', () => {
      expect(headers.find('Header').exists()).toBe(true);
    });

    it('renders a SubHeader component', () => {
      expect(headers.find('SubHeader').exists()).toBe(true);
    });

  })
});
