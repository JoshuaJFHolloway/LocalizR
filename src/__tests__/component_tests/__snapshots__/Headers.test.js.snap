// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headers Rendering components renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Headers />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Header />,
        <img
          alt="themeimage"
          src="https://cdn2.iconfinder.com/data/icons/Flag/134/Spain.png"
          width="150"
/>,
        <SubHeader />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "themeimage",
          "src": "https://cdn2.iconfinder.com/data/icons/Flag/134/Spain.png",
          "width": "150",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Header />,
          <img
            alt="themeimage"
            src="https://cdn2.iconfinder.com/data/icons/Flag/134/Spain.png"
            width="150"
/>,
          <SubHeader />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "themeimage",
            "src": "https://cdn2.iconfinder.com/data/icons/Flag/134/Spain.png",
            "width": "150",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
